ArcMovies/Modules/MovieSearch/View/MovieSearchViewController.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  MovieSearchViewController.swift
+//  ArcMovies
+//
+//  Created by Yuri Saboia Felix Frota on 22/01/18.
+//  Copyright © 2018 YuriFrota. All rights reserved.
+//
+
+import UIKit
+
+class MovieSearchViewController: UIViewController, SearchInterface {
+    var presenter: SearchPresenter!
+    var table: UITableView!
+    let searchController = UISearchController(searchResultsController: nil)
+    var moviesData = [Movie]() {
+        didSet{
+            table.reloadData()
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+    }
+    
+    fileprivate func setupView() {
+        table = TableCreator.createTable(onView: self.view)
+        table.dataSource = self
+        table.delegate = self
+        table.separatorStyle = .singleLine
+        table.tableFooterView = UIView()
+        table.rowHeight = 124
+        
+        table.register(MovieSearchTableViewCell.self)
+        
+        navigationItem.title = "Search Movies"
+        self.preferLargeTitles()
+        
+        //Searchbar
+        searchController.searchResultsUpdater = self
+        searchController.searchBar.delegate = self
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.searchBar.placeholder = "Search for movies"
+        searchController.searchBar.tintColor = .white
+        searchController.searchBar.changeTextFieldBackground(to: .white, cursor: UIColor.navigationColor)
+        navigationItem.hidesSearchBarWhenScrolling = false
+        navigationItem.searchController = searchController
+        definesPresentationContext = true
+    }
+    
+    func showMoviesData(movies: [Movie]) {
+        moviesData = movies
+    }
+    
+    func nothingToFetch() {
+        if moviesData.count == 0 {
+            UIAlertController.showPassiveAlertOn(view: self, with: "Error", and: "Please check your internet connection")
```

Comment:
I got this error running the app if I search for something that doesn't exists. How would you solve it?

Reply:
It would be necessary to improve error catching on the requests. I would add a code that checked the total results on the JSON sent from the API query, and if the numbers o results was 0, I would send a noMoviesFound error, with that I would be able to provide more detailed error messages to the user.